wlct/tournaments.py

```diff
@@ -5868,14 +5868,14 @@ def pick_template_for_game(self, team1, team2, templates_list):
                 templates_list_copy.remove(entry.game.templateid)
 
         log_tournament("Template list after prune: {}".format(templates_list_copy), self)
-        while True:
-            shuffle(templates_list_copy)
-            for tid in templates_list_copy:
-                log_tournament("Picked new template out of the list: {}".format(tid), self)
-                if self.is_template_allowed(tid, team1) and self.is_template_allowed(tid, team2):
-                    log_tournament("Picked FINAL template for game: {}".format(tid), self)
-                    return tid
-
+        shuffle(templates_list_copy)
```

Comment:
This seems like it will in fact create another undesirable infinite loop, where players on the ladder are unable to get a game. If there is no template here available to the player, can we programatically remove the player from the ladder at this point in time as well?

Reply:
If no tid is returned, the list of players will reduce with each iteration in the process_new_games func and eventually end. This should be a sanity check to prevent the edge case where a player has a valid template, but some other player vetoed them.